layouts/index.html

```diff
@@ -1,12 +1,5 @@
-{{define "main"}}
-<div class="row">
-{{ partial "tag-cloud.html" . }}
-</div>
-<div class="row">
-    {{ range .Site.RegularPages.ByLastmod.Reverse }}
-    {{ .Render "thumbnail"}}
-    {{ end }}
-</div>
-
-{{ partial "contribute.html" . }}
-{{ end }}
+{{ define "main" }}
+  {{ $section_to_display := first 1000 .Site.RegularPages.ByLastmod.Reverse }}
+  {{ $.Scratch.Add "section_to_display" $section_to_display }}
```

Comment:
Even though this works, and I don't see how it should not continue to do so, concurrency-wise, but I consider this an anti-pattern. The Scratch is built for a particular flaw in the Golang templates, and this isn't it. Pass the values you need as arguments (use dict) to the partial.